terraform/diff.go

```diff
@@ -373,6 +373,11 @@ type InstanceDiff struct {
 	// mean to be used for additional data a resource may want to pass through.
 	// The value here must only contain Go primitives and collections.
 	Meta map[string]interface{}
+
+	// Stub should be set when this diff exists only for purposes of providing a
+	// diff to various pre-plan or dry-run steps in the graph. A diff with this
+	// enabled should not be acted on in the plan.
+	Stub bool
```

Comment:
It feels a little weird to let this detail escape out into the count hook. Could we instead make `EvalDiff` skip calling the hook altogether if `Stub` is set on the eval node itself? Off the top of my head I can't see a reason why a hook would need to know about these stub diffs.